### ISSUE TITLE
Quickstart that actually works

### ISSUE THREAD

**Issue by Arkanosis** [REGOVAR/regovar-server#16]
Some stuff is probably not required though, and I'm still facing issues #14 and #15.

    $ lxc-create -n regovar_annso -t download -- -d ubuntu -r xenial -a amd64
    $ lxc-start -n regovar_annso
    $ lxc-attach -n regovar_annso
    # apt update && apt upgrade
    # apt install git ca-certificates nginx uwsgi postgresql postgresql-contrib postgresql-server-dev-9.5 build-essential libssl-dev libffi-dev python3-dev virtualenv libpq-dev libmagickwand-dev
    # useradd annso --create-home
    # sudo -u postgres createuser -P -s annso # type "annso" as password
    # sudo -u postgres createdb annso
    # mkdir -p /var/regovar/annso/{cache,downloads,files}
    # chown -R annso:annso /var/regovar/annso
    # su annso
    $ git clone https://github.com/REGOVAR/Annso.git ~/Annso
    $ cd ~/Annso
    $ virtualenv -p /usr/bin/python3.5 venv
    $ source venv/bin/activate
    $ pip install -r requirements.txt
    $ psql -U annso -d annso -f ./annso/database/create_all.sql
    $ psql -U annso -d annso -f ./annso/database/scripts/import_refgen.sql
    $ cd annso
    $ make app &!
    $ exit
    # echo 'upstream aiohttp_annso
    {
            server 127.0.0.1:8100 fail_timeout=0;
    }
    
    server
    {
            listen 80;
            listen [::]:80;
    
            location / {
                    # Need for websockets
                    proxy_http_version 1.1;
                    proxy_set_header Upgrade $http_upgrade;
                    proxy_set_header Connection "upgrade";
    
                    proxy_set_header Host $http_host;
                    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                    proxy_redirect off;
                    proxy_buffering off;
                    proxy_pass http://aiohttp_annso;
            }
    
            location /static {
                    root /var/regovar/annso;
            }
    }' > /etc/nginx/sites-available/regovar
    # rm /etc/nginx/sites-enabled/default
    # ln -s /etc/nginx/sites-available/regovar /etc/nginx/sites-enabled
    # /etc/init.d/nginx restart
    # exit
    $ IP=$(lxc-info -n regovar_annso | grep IP | sed 's/.* //')
    $ firefox $IP

I plan to put a better version of this in the documentation, and boostrap a saltstack recipe, but feel free to improve on it :)

**Comment by Oodnadatta** [comment]
I have just done two little modifications in the order of the commands above, but it works! I am now blocked by #15.

**Comment by Jnsll** [comment]
When doing `# apt update && apt upgrade`, i get the following:

```
Err:1 http://archive.ubuntu.com/ubuntu xenial InRelease
  Temporary failure resolving 'archive.ubuntu.com'
Err:2 http://security.ubuntu.com/ubuntu xenial-security InRelease
  Temporary failure resolving 'security.ubuntu.com'
Err:3 http://archive.ubuntu.com/ubuntu xenial-updates InRelease
  Temporary failure resolving 'archive.ubuntu.com'
Reading package lists... Done
Building dependency tree       
Reading state information... Done
All packages are up to date.
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/xenial/InRelease  Temporary failure resolving 'archive.ubuntu.com'
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/xenial-updates/InRelease  Temporary failure resolving 'archive.ubuntu.com'
W: Failed to fetch http://security.ubuntu.com/ubuntu/dists/xenial-security/InRelease  Temporary failure resolving 'security.ubuntu.com'
W: Some index files failed to download. They have been ignored, or old ones used instead.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

**Comment by ikit** [comment]
it seems that you don't have internet connection in your container.

**Comment by Arkanosis** [comment]
@JuneBS Try to add something like this in your `~/.confix/lxc/default.conf`:

    lxc.network.type = veth
    lxc.network.link = lxcbr0
    lxc.network.flags = up
    lxc.network.hwaddr = 00:16:3e:xx:xx:xx

    lxc.id_map = u 0 100000 65536
    lxc.id_map = g 0 100000 65536

That's a simple setup for unprivileged containers to work on Ubuntu. That should fix your issues for every LXC container you create then.

**Comment by ikit** [comment]
quickstart working on Xenial-Ubuntu

**Comment by ikit** [comment]
I change just the name of the nginx available script file

    $ lxc-create -n regovar_annso -t download -- -d ubuntu -r xenial -a amd64
    $ lxc-start -n regovar_annso
    $ lxc-attach -n regovar_annso
    # apt update && apt upgrade
    # apt install git ca-certificates nginx uwsgi postgresql postgresql-contrib postgresql-server-dev-9.5 build-essential libssl-dev libffi-dev python3-dev virtualenv libpq-dev libmagickwand-dev
    # useradd annso --create-home
    # sudo -u postgres createuser -P -s annso # type "annso" as password
    # sudo -u postgres createdb annso
    # mkdir -p /var/regovar/annso/{cache,downloads,files}
    # chown -R annso:annso /var/regovar/annso
    # su annso
    $ git clone https://github.com/REGOVAR/Annso.git ~/Annso
    $ cd ~/Annso
    $ virtualenv -p /usr/bin/python3.5 venv
    $ source venv/bin/activate
    $ pip install -r requirements.txt
    $ pip install -r requirements.txt
    $ psql -U annso -d annso -f ./annso/database/create_all.sql
    $ psql -U annso -d annso -f ./annso/database/scripts/import_refgen.sql
    $ cd annso
    $ make app &!
    $ exit
    # echo 'upstream aiohttp_annso
    {
        server 127.0.0.1:8100 fail_timeout=0;
    }

    server
    {
        listen 80;
        listen [::]:80;

        location / {
            # Need for websockets
            proxy_http_version 1.1;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "upgrade";

            proxy_set_header Host $http_host;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_redirect off;
            proxy_buffering off;
            proxy_pass http://aiohttp_annso;
        }

        location /static {
            root /var/regovar/annso;
        }
    }' > /etc/nginx/sites-available/annso
    # rm /etc/nginx/sites-enabled/default
    # ln -s /etc/nginx/sites-available/annso /etc/nginx/sites-enabled
    # /etc/init.d/nginx restart
    # exit
    $ IP=$(lxc-info -n regovar_annso | grep IP | sed 's/.* //')
    $ firefox $IP

**Comment by ikit** [comment]
I close this issue. 
The script is now available here : https://github.com/REGOVAR/Annso/blob/master/docs/installation_script
Thanks to open new issue if you encounter a problem with the script.
